ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -175,25 +180,23 @@ private[kvutils] case class ProcessTransactionSubmission(
     for {
       damlState <- getDamlState
       startingKeys = damlState.collect {
-        case (k, v) if k.hasContractKey && v.getContractKeyState.hasContractId => k
+        case (k, v) if k.hasContractKey && v.getContractKeyState.nonEmpty => k
       }.toSet
 
       allUnique = relTx
         .fold((true, startingKeys)) {
           case (
               (allUnique, existingKeys),
-              (_nodeId, exe: NodeExercises[_, _, VersionedValue[ContractId]]))
+              (_nodeId, exe @ Node.NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _)))
               if exe.key.isDefined && exe.consuming =>
             val stateKey = Conversions.globalKeyToStateKey(
-              GlobalKey(exe.templateId, Conversions.forceNoContractIds(exe.key.get.key.value)))
+              Node.GlobalKey(exe.templateId, Conversions.forceNoContractIds(exe.key.get.key.value)))
```

Comment:
we can drop the `forceNoContractIds` as the GlobalKey constructor hashes it, which will throw if there's contract ids.

Reply:
fixed